api/cloud/oc_cloud.c

```diff
@@ -109,16 +111,25 @@ cloud_deregister_on_reset_internal(oc_cloud_context_t *ctx,
 {
   (void)status;
   (void)data;
+  oc_cloud_clear_context(ctx);
+}
+#endif /* OC_SECURITY */
+
+void
+oc_cloud_clear_context(oc_cloud_context_t *ctx)
+{
+  oc_assert(ctx != NULL);
+
   cloud_close_endpoint(ctx->cloud_ep);
   memset(ctx->cloud_ep, 0, sizeof(oc_endpoint_t));
   ctx->cloud_ep_state = OC_SESSION_DISCONNECTED;
   cloud_store_initialize(&ctx->store);
+  cloud_store_dump(&ctx->store);
   cloud_manager_stop(ctx);
   ctx->last_error = 0;
   ctx->store.cps = 0;
   cloud_store_dump(&ctx->store);
```

Comment:
We now have 2 `cloud_store_dump` calls here, which is surely wrong. The `cps` value is used in `cloud_store_dump`, so I think the second call (after `cps` is assigned 0) makes more sense. (Though I recall that I copied that code from somewhere and it had only the _first_ call to `cloud_store_dump`)

Is is deliberate that `cloud_store_initialize` doesn't set `cps` to 0?

Reply:
From my point of view, `cps` must be set to 0 at `cloud_store_initialize`.
@WAvdBeek what do you think ?